### DISCUSSION TITLE
How can we implement Multi-tenancy and qlight in GoQuorum?

### DISCUSSION THREAD

**Discussion by ghost** [ConsenSys/quorum#1498] [Q&A] [answered]
I am trying to implement Multi-tenancy and qlight in GoQuorum. So please provide sample code example with steps.

Thanks in advanced!

**Comment by antonydenyer** [comment, marked as answer]
https://consensys.net/docs/goquorum/en/latest/configure-and-manage/configure/qlight-node/ could let us know which part you're struggling with? thanks

**Reply by ghost** [reply]
Hi Antony, I am still stuck in Multi-tenancy.

Below are the steps which I have done:

Added (enableMultiplePrivateStates and residentGroups) in congig.json file of Tessera Node1.
Added ( "isMPS": true) in genesis file of Node1.
And now Tessera Node1 is running but when i am trying to run the Node1 then i am getting Fatal error:

Error -- Fatal: multitenancy requires RPC Security Plugin to be configured

So how can I implement RPC Security Plugin ?

**Reply by ghost** [reply]
Could you please share any example or sample code if any ?

**Reply by antonydenyer** [reply]
There are examples on how to configure the plugin at https://github.com/ConsenSys/quorum-security-plugin-enterprise/tree/master/examples

What the error message is saying is the you need to have some form of external security running so that the node knows which tenant to serve up.  

You should probably start with multi-tenancy first then qlight.

**Reply by ghost** [reply]
Thanks for your reply! So can i use  https://github.com/ConsenSys/quorum-security-plugin-enterprise/tree/master/examples for end to end implementation of Multi-tenancy because its very old repository (2 yrs old).?

**Reply by antonydenyer** [reply]
Yes it's still active.

**Reply by ghost** [reply]
ok.. thanks.

Will try this and will let you know for any further queries.

**Reply by ghost** [reply]
Hi Antony, Actually i am starting Quorum and the Authorization Server using this command -docker-compose -f quorum.yml -f hydra.yml up -d 

But All nodes are getting shutdown while connecting Tessera node. 
<img width="943" alt="Containers-screenshot" src="https://user-images.githubusercontent.com/35056245/189825613-eefe3a39-3673-4c72-8fa7-8b5132f74bc9.PNG">

Error is:

<img width="776" alt="quorum-error" src="https://user-images.githubusercontent.com/35056245/189826078-4466b097-548e-4b22-b9d4-d957ddd3c03b.PNG">

--------------------
Kindly help me on this issue.

**Reply by antonydenyer** [reply]
You're running an old version of tessera that doesn't support multitenancy.

**Reply by ghost** [reply]
Ok got it. but when I am trying to up the yaml using latest version of tessera getting below error in endpoint path: 
(/tessera/tessera-app.jar not found).

So please let me know how many changes would be required in yaml?

**Reply by ghost** [reply]
Below is the actual error:
Error: Unable to access jarfile /tessera/tessera-app.jar

**Reply by ghost** [reply]
And now am getting below error while running quorum.yml with tessera version 21.7.2..

Error: Invalid or corrupt jarfile /tessera/bin/tessera

Please help me on this issue?

**Reply by baptiste-b-pegasys** [reply]
Hi, new version of Tessera is launched differently.
You can check how it is done here: https://github.com/ConsenSys/quorum-examples/blob/master/examples/7nodes/tessera-start.sh#L35 or in the docker file https://github.com/ConsenSys/quorum-examples/blob/master/docker-compose.yml#L207
